src/platform/windows/misc.cpp

```diff
@@ -200,6 +202,71 @@ namespace platf {
     return std::string(buffer, bytes);
   }
 
+  /**
+ * @brief A function to duplicate the current sessions user's token with elevated privileges
+ *
+ * @return A handle to the duplicated users token, or null if the duplication failed
+ */
+  HANDLE
+  duplicate_users_token_elevated() {
```

Comment:
You can change this to accept a parameter for whether it wants the elevated token or not, then you can get rid of `duplicate_shell_token()` and use this function for both cases.